lib/AktiveMerchant/Billing/Gateways/Trexle.php

```diff
@@ -0,0 +1,442 @@
+
+    /**
+     * {@inheritdoc}
+     */
+    public static $display_name = 'Trexle';
+
+    /**
+     * {@inheritdoc}
+     */
+    public static $default_currency = 'USD';
+
+    /**
+     * Additional options needed by gateway
+     *
+     * @var array
+     */
+    protected $options;
+
+    /**
+     * Contains the main body of the request.
+     *
+     * @var array
+     */
+    private $post;
+
+    /**
+     * Creates gateway instance from given options.
+     *
+     * @param array $options An array contains login parameters of merchant
+     *                       and optional currency.
+     */
+    public function __construct($options = array())
+    {
+        Options::required('secret_key', $options);
+
+        parent::__construct($options);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function authorize($money, CreditCard $creditcard, $options = array())
+    {
+        $this->post = array();
+        $options = new Options($options);
+
+        $this->addInvoice($money, $options);
+        $this->addCreditCard($creditcard);
+        if (null === $creditcard->token) {
+            $this->addAddress($options);
+        }
+        $this->addCustomerData($options);
+
+        $this->post['capture'] = 'false';
+
+        return $this->commit(self::CHARGE);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function purchase($money, CreditCard $creditcard, $options = array())
+    {
+        $this->post = array();
+        $options = new Options($options);
+
+        $this->addInvoice($money, $options);
+        $this->addCreditCard($creditcard);
+        if (null === $creditcard->token) {
+            $this->addAddress($options);
+        }
+        $this->addCustomerData($options);
+
+        return $this->commit(self::CHARGE);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function capture($money, $authorization, $options = array())
+    {
+        $this->post = array('amount' => $this->amount($money / 100));
+
+        $action = sprintf(self::CAPTURE, $authorization);
+
+        return $this->commit($action, RequestInterface::METHOD_PUT);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function void($authorization, $options = array())
+    {
+        $this->post = array();
+        $action = sprintf(self::REFUND, $identification);
+
+        return $this->commit($action);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function credit($money, $identification, $options = array())
+    {
+        $this->post = array('amount' => $this->amount($money / 100));
+
+        $action = sprintf(self::REFUND, $identification);
+
+        return $this->commit($action);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * If you want to unstore a credit card you should also keep card token from
+     * response `$response->card->token`
+     * Then pass it as option in unstore method below.
+     */
+    public function store(CreditCard $creditcard, $options = array())
```

Comment:
Looks all good to me know. Do you want me to add a /tests/AktiveMerchant/Billing/Gateways/TrexleTest.php file?